### PR TITLE
WireGuard support

### PR DESCRIPTION
**Pull request by alvra**
Implement WireGuard support.

### CODE REVIEW

**Review by gijzelaerr — CHANGES_REQUESTED** [review]
I can't get wireguard to work, see chat discussion in slack for details.

2 issues:
 * I can't connect to `vpn-next.tuxed.net` and I don't know why, I don't get an error.
 * If I then connect to a different VPN server a need to select a profile, the list here seems to be the `vpn-next.tuxed.net` list. If I select one I get the error:
 
```
Traceback (most recent call last):
  File "/home/gijs/Work/surfnet/python-eduvpn-client/eduvpn/interface/event.py", line 165, in on_chosen_profile
    connection = connect_server_info.connect(app, profile, oauth_session)
  File "/home/gijs/Work/surfnet/python-eduvpn-client/eduvpn/server.py", line 261, in connect
    remote.check_response(response)
  File "/home/gijs/Work/surfnet/python-eduvpn-client/eduvpn/remote.py", line 52, in check_response
    uri = response.history[0].url
IndexError: list index out of range
```

File: eduvpn/utils.py [hunk 1]
```diff
@@ -8,13 +8,20 @@
...
+from requests.adapters import HTTPAdapter, Retry
 
 logger = getLogger(__file__)
 
 
 def get_logger(name_space: str):
     return getLogger(name_space)
 
+def add_retry_adapter(session: Session, retries: int):
+        adapter = HTTPAdapter(max_retries=Retry(total=retries))
```
Comment (alvra) [inline eduvpn/utils.py:21]: Please use 4 space indentation

Comment (jwijenbergh) [reply eduvpn/utils.py:21]: whoops, thanks

File: eduvpn/nm.py [hunk 2]
```diff
@@ -293,12 +293,19 @@ def start_wireguard_connection(
...
+        stripped_entry = dns_entry.strip()
+        try:
+            address = ip_address(stripped_entry)
+            if address.version == 4:
+                dns4.append(str(address))
+            elif address.version == 6:
+                dns6.append(str(address))
+        # The entry is not an ip but a hostname
+        # They need to be added to dns search domains
+        except ValueError:
```
Comment (alvra) [inline eduvpn/nm.py:307]: I prefer to keep the try branch as short as possible, with an `else` branch for the rest.

**Comment by alvra** [comment]
Great addition @jwijenbergh!

**Pull request merged**